index.go

```diff
@@ -119,6 +119,12 @@ func (b *Batch) Reset() {
 	b.internal.Reset()
 }
 
+func (b *Batch) Merge(o *Batch) {
+	b.internal.Merge(o.internal)
+	b.lastDocSize = o.LastDocSize()
+	b.totalSize += o.TotalDocsSize()
```

Comment:
From a quick first-gut reaction, not sure if it's right for total docs size to be simply summed this way.  For example, if the 2nd batch had updates or deletions of docs of the same key(s) as from the 1st batch, then the ultimate size might be different?

Reply:
you are right, thanks!